enterprise/internal/own/background/analytics.go

```diff
@@ -18,33 +20,44 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
-func handleAnalytics(ctx context.Context, logger log.Logger, repoId api.RepoID, db database.DB) error {
+func handleAnalytics(ctx context.Context, lgr log.Logger, repoId api.RepoID, db database.DB, subRepoPermsCache *rcache.Cache) error {
 	// 🚨 SECURITY: we use the internal actor because the background indexer is not associated with any user,
 	// and needs to see all repos and files.
 	internalCtx := actor.WithInternalActor(ctx)
-	indexer := newAnalyticsIndexer(gitserver.NewClient(), db)
-	err := indexer.indexRepo(internalCtx, repoId)
+	indexer := newAnalyticsIndexer(gitserver.NewClient(), db, subRepoPermsCache, lgr)
+	err := indexer.indexRepo(internalCtx, repoId, authz.DefaultSubRepoPermsChecker)
 	if err != nil {
-		logger.Error("own analytics indexing failure", log.String("msg", err.Error()))
+		lgr.Error("own analytics indexing failure", log.String("msg", err.Error()))
 	}
 	return err
 }
 
 type analyticsIndexer struct {
-	client gitserver.Client
-	db     database.DB
+	client            gitserver.Client
+	db                database.DB
+	logger            log.Logger
+	subRepoPermsCache *rcache.Cache
 }
 
-func newAnalyticsIndexer(client gitserver.Client, db database.DB) *analyticsIndexer {
-	return &analyticsIndexer{client: client, db: db}
+func newAnalyticsIndexer(client gitserver.Client, db database.DB, subRepoPermsCache *rcache.Cache, lgr log.Logger) *analyticsIndexer {
+	return &analyticsIndexer{client: client, db: db, subRepoPermsCache: subRepoPermsCache, logger: lgr}
```

Comment:
I'd rather encapsulate cache initialization, because it is more of an implementation detail which is not that important to callers.
Another benefit of doing this, is that we can remove nil checks and `noCache` variable checks in `isSubRepoPermsRepo` function.

WDYT?

```suggestion
func newAnalyticsIndexer(client gitserver.Client, db database.DB, lgr log.Logger) *analyticsIndexer {
	return &analyticsIndexer{client: client, db: db, subRepoPermsCache: rcache.New("analytics"), logger: lgr}
```

Reply:
This applies to all indexers :)

Reply:
ha, but we have the cache always set in `background.go`.
Why do we have the checks, then? Should we just remove them?

Reply:
I removed the nil case for the cache in my latest commit.

How do you feel about sharing the cache bw the indexers?